forge/forge-policy.json

```
@@ -1,95 +1,162 @@
-              "fsx:TagResource",
-              "ec2:DescribeSecurityGroups",
-              "ec2:DescribeAccountAttributes",
-              "ec2:DescribeSubnets",
-              "ec2:DescribeLaunchTemplates",
-              "ec2:DescribeLaunchTemplateVersions", 
-              "ec2:CreateLaunchTemplate",
-              "ec2:DeleteLaunchTemplate",
-              "ec2:DescribeKeyPairs",
-              "ec2:DescribeVpcs",
-              "ec2:DescribeInstanceTypeOfferings",
-              "ec2:GetEbsEncryptionByDefault",
-              "elasticfilesystem:DescribeMountTargets",
-              "elasticfilesystem:CreateMountTarget",
-              "elasticfilesystem:CreateFileSystem",
-              "elasticfilesystem:DescribeFileSystems",
-              "elasticfilesystem:DeleteMountTarget",
-              "elasticfilesystem:DeleteFileSystem",
-              "elasticfilesystem:UpdateFileSystem",
-              "elasticfilesystem:PutLifecycleConfiguration",
-              "elasticfilesystem:TagResource"
-          ],
-          "Resource": "*"
-      },
-      {
-          "Sid": "TowerLaunch0",
-          "Effect": "Allow",
-          "Action": [
-              "s3:Get*",
-              "s3:List*",
-              "batch:DescribeJobQueues",
-              "batch:CancelJob",
-              "batch:SubmitJob",
-              "batch:ListJobs",
-              "batch:TagResource",
-              "batch:DescribeComputeEnvironments",
-              "batch:TerminateJob",
-              "batch:DescribeJobs",
-              "batch:RegisterJobDefinition",
-              "batch:DescribeJobDefinitions",
-              "ecs:DescribeTasks",
-              "ec2:DescribeInstances",
-              "ec2:DescribeInstanceTypes",
-              "ec2:DescribeInstanceAttribute",
-              "ecs:DescribeContainerInstances",
-              "ec2:DescribeInstanceStatus",
-              "ec2:DescribeImages",
-              "logs:Describe*",
-              "logs:Get*",
-              "logs:List*",
-              "logs:StartQuery",
-              "logs:StopQuery",
-              "logs:TestMetricFilter",
-              "logs:FilterLogEvents",
-              "ses:SendRawEmail"
-          ],
-          "Resource": "*"
-      }
+    {
+      "Sid": "SSMParameterAccess",
+      "Effect": "Allow",
+      "Action": "ssm:GetParameters",
+      "Resource": "*"
+    },
+    {
+      "Sid": "IAMRoleAndProfileManagement",
+      "Effect": "Allow",
+      "Action": [
+        "iam:CreateInstanceProfile",
+        "iam:DeleteInstanceProfile",
+        "iam:AddRoleToInstanceProfile",
+        "iam:RemoveRoleFromInstanceProfile",
+        "iam:CreateRole",
+        "iam:DeleteRole",
+        "iam:GetRole",
+        "iam:AttachRolePolicy",
+        "iam:DetachRolePolicy",
+        "iam:PutRolePolicy",
+        "iam:DeleteRolePolicy",
+        "iam:ListAttachedRolePolicies",
+        "iam:ListRolePolicies",
+        "iam:TagRole",
+        "iam:TagInstanceProfile"
+      ],
+      "Resource": "*"
```

By default unless the prefix is changed we could restrict this to IAM roles starting with `TowerForge-`, and/or add a reference to the readme above in the docs section where we explain the custom forge prefix

---

```suggestion
      "Resource": "arn:aws:iam::*:role/TowerForge-*",
```

This would limit iam roles access to only roles starting with the default prefix `TowerForge-`, but in the readme we should explain to change it if customers set a custom prefix
this would make the policy less generic and less plug-and-play though

---

Let's keep the policy as generic as possible, comments were left in the readme on how to lock it down further.